src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -144,14 +144,19 @@ private void OnFileChanged(FileSystemEventArgs e)
             {
                 // app_offline.htm has changed
                 // when app_offline.htm is created, we trigger
-                // a shutdown so when the host starts back up it
-                // will be offline
+                // a shutdown right away so when the host
+                // starts back up it will be offline
                 // when app_offline.htm is deleted, we trigger
                 // a restart to bring the host back online
+                // we also need to remove the debounce time here
+                // to avoid race condition causing the Host to
+                // not shutdown before a deployment starts
                 changeDescription = "File";
                 if (File.Exists(e.FullPath))
                 {
-                    shutdown = true;
+                    _logger.LogInformation($"{changeDescription} change of type {e.ChangeType.ToString()} detected for '{e.Name}'");
```

Comment:
Changes look good, but can we refactor this to avoid the code/message duplication with what we have in `TraceFileChangeRestart`?